fix angular draggable styling

huayunh opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes #8 , identified #32 

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
Changes proposed in this Pull Request:
- `cdk-drag-placeholder` no longer visible
- added a swapping animation to make it less stiff
- added a couple comments

joebochill commented on src/app/app.component.scss:
```diff
@@ -2,20 +2,30 @@
+    position: relative;
+    height: inherit;
+    margin-left: 4px;
+    border-radius: 4px;
```
This can be reduced to:
```
display: flex;
align-items: center;
padding: 0 16px;
background-color: white;
```

joebochill commented on src/app/app.component.scss:
```diff
@@ -5,19 +5,10 @@
+    display: flex;
+    align-items: center;
+    padding: 0 16px;
+    background-color: white;
```
Didn't mean for you to remove the box-shadow...github range comments are not very clear about what is highlighted :(

huayunh commented:
For the two commits above:
1. Added drop shadow back
2. fixed the misalignment issue for `cdk-drag-preview`
3. I noticed everytime the user enters the editing mode by clicking the pencil icon on the tool bar, the height of `mat-list-item` changes from 48px to 56px. Angular branch is the only one having this problem, so I guess this is not a concerted behavior. I discovered that this was due to the `<mat-icon>` used only in the editing mode, so I added another `mat-icon` on line 11 and made it invisible.

joebochill commented on src/app/app.component.html:
```diff
@@ -7,22 +7,23 @@
 	<section  *ngIf="!draggable">
-		<mat-list-item   *ngFor="let i of data">
+		<mat-list-item *ngFor="let i of data">
+			<mat-icon style="display: none;" matListIcon cdkDragHandle class="handler">drag_indicator</mat-icon>
```
Adding an icon here messes up the alignment, and we really shouldn't be using invisible elements to do layout. Instead, just set a style rule to put a minHeight on the rows (56).

huayunh replied:
It does not mess up with alignment, and using min-height would need some use of `!important` in css, which I personally prefer not to have.

joebochill commented on src/app/app.component.scss:
```diff
@@ -19,4 +23,17 @@
+.cdk-drag-preview div.details {
+  position: absolute;
+  right: 0;
+}
```
I'm not a fan of using position absolute for things like this, but I haven't been able to find a better workaround, so it's ok.

joebochill commented:
The issue seems to be around using the mat-list-item in a cdk drag/drop environment. The layout of the list-item changes when it becomes draggable. Seems to be a known issue in Angular Material. If you use something else, like a simple <div> it works perfectly, as expected.
